gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/filter/CoinStep.java

```diff
@@ -73,7 +85,7 @@ public String toString() {
 
     @Override
     public int hashCode() {
-        return super.hashCode() ^ Double.hashCode(this.probability);
+        return super.hashCode() ^ Double.hashCode(this.probability.get());
```

Comment:
can `Double.hashCode(this.probability.get())` be replaced with `this.probability.hashCode()` ?

Reply:
Those will not be equivalent as GValue.hashCode() combines the hash of the name, type, and value. I believe it's best if the equality of CoinStep is based on the value of the probability, not the name of any parameter.